### PR TITLE
Clonerefs and image-builder tasks for tekton

### PR DESCRIPTION
**Pull request by dekiel**
**Description**

Changes proposed in this pull request:

- Added external-secrets secrets store resource definition file.
- Added external-secrets operator values file with workload identity service account annotations.
- Added external-secrets secrets resources definitions. Secrets are required by demo prowjob using pipelines.
- Added wrapper around k8s test-infra clonerefs tool to let it run as Tekton Task.
- Added Dockerfile to buid clonerefs for Tekton pipeline image.
- Added clonerefs Task defintion using clonerefs for Tekton.
- Added build-image Task definition using kyma test-infra image-builder tool.
- Added demo prowjob running as Tekton pipeline.

See #6798

### CODE REVIEW

**Comment by Ressetkk** [comment]
What was the problem with `git-clone` task that has been added initially during set up? It's natively maintained by Tekton devs and seemed to be easy enough to use instead of creating a wrapper around Prow's clonerefs.

Image-builder was not meant to be used in workloads other than Prow, so YMMV. I would be cautious implementing that in Tekton because you won't have easy ability to build for example local images.

**Comment by dekiel** [comment]
The problem with `git-clone` is that it doesn't integrate easily with prow. With clonerefs, a developer has more prow like behaviour they used to.

This is possible because clonerefs uses variables set by prow to clone refs defined in prowjobs. Same as kubernetes agent clonerefs does.
Because by default it clone refs in the same location as kubernetes agent clonerefs does.

This makes it easier to use by our developers. This makes easier to reuse existing tools and scripts in tekton pipelines.

What are local images?
What is the problem with building these local images on tekton?
How building local images is solved when running on k8s agent?
I did test image-builder few times and it's able to build image, sign it and push it to the registry. A cache works as well. After all in both scenarios it's running as a k8s pod.

File: prow/jobs/test-infra/tekton-demo.yaml [hunk 1]
```diff
@@ -0,0 +1,146 @@
+presubmits:
+  kyma-project/test-infra:
+    - name: tekton-demo
```
Comment (Ressetkk) [inline prow/jobs/test-infra/tekton-demo.yaml:3]: Please remove this file, or move it as an example to the docs. If it's a demo job this should not be added to the Prow.

File: task/clone-refs/0.1/README.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
+# Unit Tests for Go Code
```
Comment (Ressetkk) [inline task/clone-refs/0.1/README.md:1]: Wrong title

File: task/build-image/0.1/README.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
+# Image Builder
+
+The image-builder Task builds an OCI image from a Dockerfile.
```
Comment (Ressetkk) [inline task/build-image/0.1/README.md:3]: There's inconsistency in naming convention for image-builder. Additionally I think this should be stated that this is a task in the title

File: task/clone-refs/0.1/README.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+The clonerefs Task clone repositories references defined by prowjob refs and extra-refs.
```
Comment (Ressetkk) [inline task/clone-refs/0.1/README.md:3]: again, there's inconsistency in naming convention. The directory is called `clone-refs` but you refer it as `clonerefs` here

**Review by Ressetkk — CHANGES_REQUESTED** [review]

File: task/build-image/0.1/README.md (hunk 3 above)
Comment (dekiel) [reply task/build-image/0.1/README.md]: This is a task documentation. No need to repeat that in the title. Aligned with other docs

File: tekton/deployments/external-secrets/values.yaml [hunk 5]
```diff
@@ -0,0 +1,390 @@
...
+  create: true
+  # -- Specifices the time to check if the cert is valid
+  certCheckInterval: "5m"
+  # -- Specifices the lookaheadInterval for certificate validity
+  lookaheadInterval: ""
+  replicaCount: 1
+  certDir: /tmp/certs
+  # -- specifies whether validating webhooks should be created with failurePolicy: Fail or Ignore
+  failurePolicy: Fail
+  # -- Specifies if webhook pod should use hostNetwork or not.
```
[the remark this answers is not in the document]
Comment (dekiel) [reply tekton/deployments/external-secrets/values.yaml:164]: All comments in tekton/deployments/external-secrets/values.yaml file are original comments existing in upstream values.yaml file. There is no sense to change them. They should be keep in original version.

File: task/clone-refs/main.go [hunk 6]
```diff
@@ -0,0 +1,88 @@
+// This is a usage of clonerefs tool from kubernetes/test-infra repo adjusted to work with Tekton as a Task.
```
[the remark this answers is not in the document]
Comment (dekiel) [reply task/clone-refs/main.go:1]: An original tool name, coming from Kubernetes team is clonerefs, without dash in the middle.

File: tekton/deployments/external-secrets/external_secrets.yaml [hunk 7]
```diff
@@ -0,0 +1,54 @@
+---
+apiVersion: external-secrets.io/v1beta1
+kind: ExternalSecret
+metadata:
+  name: signify-prod-secret # name of the k8s external secret and the k8s secret
```
[the remark this answers is not in the document]
Comment (dekiel) [reply tekton/deployments/external-secrets/external_secrets.yaml:5]: All comments in external_secrets.yaml file are coming from original file from upstream repository. They should not be changed. We use the same exact copy in other files defining this type of resoource.

File: task/image-builder/0.1/README.md [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+- **`JOB_TYPE`**: Type of prowjob. Variable set by prow according. _(REQUIRED)_
+- **`PULL_NUMBER`**: Pull request number. Variable set by prow. _(REQUIRED)_
+- **`PULL_BASE_SHA`**: Git SHA of the base branch. Variable set by prow. _(REQUIRED)_
+- **`CI`**: Set to true when the current environment is a CI environment. Variable set by prow. _(REQUIRED)_
+- **`REPO_OWNER`**: GitHub org that triggered the job. Variable set by prow. _(REQUIRED)_
+- **`REPO_NAME`**: GitHub repo that triggered the job. Variable set by prow. _(REQUIRED)_
+
+## Platforms
+
+The Pipeline can be run on `linux/amd64` platform.
```
[the remark this answers is not in the document]
Comment (dekiel) [reply task/image-builder/0.1/README.md:42]: This should mention Task not Pipeline. Corrected.

File: task/clone-refs/0.1/README.md [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+## Parameters
+
+- **`JOB_SPEC`**: JSON-encoded job specification. Variable set by prow. _(REQUIRED)_
+- **`LOG`**: The path to the clone logs file. _(OPTIONAL, default: "/logs/clone.json")_
+  prowjob type is periodic. In that case an empty string default value will be used. _(OPTIONAL, default: "")_
+- **`SRC_ROOT`**: The root path where to clone repositories. _(OPTIONAL, default: "/home/prow/go")_
+
+## Platforms
+
+The Pipeline can be run on `linux/amd64` platform.
```
[the remark this answers is not in the document]
Comment (dekiel) [reply task/clone-refs/0.1/README.md:35]: This should refer to Task not Pipeline. Corrected.

File: docs/examples/tekton-demo.yaml [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+                type: string
+            # The following resources are not used by the pipeline, but are required by the ProwJob syntax validator.
+            # They are not used by the pipeline because the pipeline uses environment variables to get extra_refs to clone the repo in a task.
+            # This is a workaround for deprecated resource object in Tekton Pipeline api version v1beta1.
+            resources:
+              - name: extra-ref
+                type: git
+            description: "Tekton Demo"
+            # TaskRuns executed by teh pipieline. TaskRuns provide values and implementation details required by Tasks executed by the pipeline.
+            # TaskRuns provide values and implementations explicitly or binding them to values and implementations provided by the pipeline.
```
[the remark this answers is not in the document]
Comment (dekiel) [reply docs/examples/tekton-demo.yaml:99]: Changed to the first suggested version.

**Review by IwonaLanger — DISMISSED** [review]

**Review by IwonaLanger — APPROVED** [review]

**Review by Ressetkk — APPROVED** [review]

**Pull request merged**